_overviews/scala3-book/scala-for-java-devs.md

```diff
@@ -122,9 +123,9 @@ This section provides comparisons of features related to OOP-style classes and m
     <tr>
       <td class="java-block">
         <code>class Person {
-          <br>&nbsp; private String firstName;
-          <br>&nbsp; private String lastName;
-          <br>&nbsp; private int age;
+          <br>&nbsp; public String firstName;
+          <br>&nbsp; public String lastName;
+          <br>&nbsp; public int age;
```

Comment:
This is not idiomatic Java but since there are no getters/setters I think it is better to make them public, to be consistent with the Scala version (or, we could make them private in the Scala version as well).

Reply:
I think I made a mistake on this. My intent was to show idiomatic Java, including the getters and setters, but forgot them for some reason. My thinking is Java developers will be reading this page, so I want to show correct Java code, i.e., what they’re used to seeing.

So I think I’d prefer to show private fields along with public getters and setters on the Java side. Then add a few words to explain the difference between that Java code and the Scala code, i.e., that Scala does not use `get` and `set` prefixes.

What do you think? (Also, if that makes sense, I can update the file if you prefer.)

Reply:
What would be the Scala code?

Reply:
Okay, sorry to flip-flop here, but I worked through the JavaBeans style code, and I don’t like going down that road. Therefore, I think I would keep the current code, and in the *OOP style class, primary constructor* section I would precede it with a little introduction like this:

“Scala doesn’t follow the JavaBeans standard, so instead of showing Java code written in the JavaBeans style, here we show Java code that is equivalent to the Scala code that follows it.”

Does something like that sound okay?

Reply:
Done.